code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -522,6 +574,41 @@ func (uar *updateAllocationRequest) getNewBlobbersSize(
 	return alloc.BlobberDetails[0].Size + uar.getBlobbersSizeDiff(alloc)
 }
 
+func (sc *StorageSmartContract) getBlobbersByIDs(ids []string, balances chainstate.StateContextI) []*StorageNode {
+	type blobberResp struct {
+		index   int
+		blobber *StorageNode
+	}
+
+	blobberCh := make(chan blobberResp, len(ids))
+	var wg sync.WaitGroup
+	for i, details := range ids {
+		wg.Add(1)
+		go func(index int, blobberId string) {
+			defer wg.Done()
+			blobber, err := sc.getBlobber(blobberId, balances)
+			if err != nil || blobber == nil {
+				logging.Logger.Debug("can't get blobber", zap.String("blobberId", blobberId), zap.Error(err))
+				return
+			}
+			blobberCh <- blobberResp{
+				index:   index,
+				blobber: blobber,
+			}
+		}(i, details)
+	}
+	wg.Wait()
+	close(blobberCh)
+
+	//ensure original ordering
+	blobbers := make([]*StorageNode, len(ids))
+	for resp := range blobberCh {
+		blobbers[resp.index] = resp.blobber
```

Comment:
If error happens in the above blobber getting gorountine, for example, blobber of index 1 failed, then `blobbers[1]` will be nil, and the caller will panic if it does not do the nil check.